src/lib/core/line/line.ts

```diff
@@ -16,15 +17,20 @@ export class MdLine {}
 
 /* Helper that takes a query list of lines and sets the correct class on the host */
 export class MdLineSetter {
+  private _changesSubscription: Subscription;
+
   constructor(private _lines: QueryList<MdLine>, private _renderer: Renderer,
               private _element: ElementRef) {
     this._setLineClass(this._lines.length);
 
-    this._lines.changes.subscribe(() => {
+    this._changesSubscription = this._lines.changes.subscribe(() => {
       this._setLineClass(this._lines.length);
     });
   }
 
+  public destroy() {
+    this._changesSubscription.unsubscribe();
+  }
```

Comment:
This particular unsubscription isn't necessary because the `QueryList` will be destroyed at the same time as the component (and the subscription is contained within the component that also contains this LineSetter).